demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1601 @@
+         * publish message back to the application. This function also
+         * sends ping request to broker if MQTT_KEEP_ALIVE_INTERVAL_SECONDS
+         * has expired since the last MQTT packet sent and receive
+         * ping responses. */
+        mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            LogWarn( ( "MQTT_ProcessLoop failed: Error = %s.",
+                       MQTT_Status_strerror( mqttStatus ) ) );
+        }
+
+        /* The humidity callback should have been invoked for handling the incoming
+         * PUBLISH message on the humidity topic. */
+        else if( globalReceivedHumidityData != true )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Registered callback was not invoked on sending a PUBLISH to "
+                        "the humidity topic: Topic=%s", DEMO_HUMIDITY_TOPIC ) );
+        }
+    }
+
+    /* Subscribe to the precipitation topic filter, this time also without any wildcard characters,
+     * so that we can receive incoming PUBLISH message only on the same topic from the broker. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Register subscription callback for the wildcard notification topic filter. */
+        ( void ) SubscriptionManager_RegisterCallback( DEMO_PRECIPITATION_TOPIC,
+                                                       DEMO_PRECIPITATION_TOPIC_LENGTH,
+                                                       precipitationDataCallback );
+
+        /* The client is now connected to the broker. Subscribe to the precipitation topic
+         * by sending a subscribe packet. This client will then publish to the same topic it
+         * subscribed to, so it will expect all the messages it sends to the broker
+         * to be sent back to it from the broker. This demo uses QOS2 in Subscribe,
+         * therefore, the Publish messages received from the broker will have QOS2. */
+        LogInfo( ( "Subscribing to the MQTT topic %s.",
+                   DEMO_PRECIPITATION_TOPIC ) );
+        returnStatus = subscribeToTopic( &mqttContext,
+                                         DEMO_PRECIPITATION_TOPIC,
+                                         DEMO_PRECIPITATION_TOPIC_LENGTH );
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Process incoming packet from the broker. Acknowledgment for subscription
+         * ( SUBACK ) will be received here. However after sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Since this
+         * demo is subscribing to the topic to which no one is publishing, probability
+         * of receiving publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet. This demo uses MQTT_ProcessLoop to
+         * receive packet from network. */
```

Comment:
This is wordy and does not need to be repeated every time we subscribe to something. A `/* Receive SUBACK from broker. */` would be sufficient here.

Reply:
Removed the wordiness